macros/src/codegen.rs

```diff
@@ -490,6 +490,43 @@ pub fn generate_code(sm: &ParsedStateMachine) -> proc_macro2::TokenStream {
         quote! {Error}
     };
 
+    let mut state_derives = Vec::new();
+
+    if sm.impl_debug_states {
+        state_derives.push(quote!{Debug});
+    }
+
+    let state_derives = if !state_derives.is_empty() {
+        quote!{#[derive(#(#state_derives),*)]}
+    } else {
+        quote!{}
+    };
+
+    let mut event_derives = Vec::new();
+
+    if sm.impl_debug_events {
+        event_derives.push(quote!{Debug});
+    }
+
+    let event_derives = if !event_derives.is_empty() {
+        quote!{#[derive(#(#event_derives),*)]}
+    } else {
+        quote!{}
+    };
+
+    let mut state_machine_derives = Vec::new();
+
+    if sm.impl_debug_state_machine {
+        state_machine_derives.push(quote!{Debug});
+    }
+
+    let state_machine_derives = if !state_machine_derives.is_empty() {
+        quote!{#[derive(#(#state_machine_derives),*)]}
+    } else {
+        quote!{}
+    };
+
+
```

Comment:
Why add a flag to the smlang macro when you can just unconditionally add `#[derive(Debug)]` onto the objects?

I believe `#[derive(Debug)]` properly accounts for generics etc. and does not impl Debug if inner attributes do not impl `Debug`

Reply:
I wonder if this is true for non-generic types though... In our case, `States` may have non-generic data that does not `impl Debug`.

In general, I'm bringing this up because the additional args in the smlang macro don't seem sustainable.